provider/backingchain/blockcommit_base.py

```
@@ -0,0 +1,181 @@
+            if not self.vm.is_alive():
+                self.vm.start()
+            self.prepare_iscsi()
+
+        # Update vm disk
+        libvirt.set_vm_disk(self.vm, self.params)
+
+    def prepare_snapshot(self, snap_num=3):
+        """
+        Prepare domain snapshot
+          :params snap_num: snapshot number, default value is 3
+
+        """
+        # Create backing chain
+        if not self.vm.is_alive():
+            self.vm.start()
+        for i in range(snap_num):
+            options = "%s --disk-only --diskspec %s,file=%s" \
+                      % ('snap%d' % i, self.new_dev, self.tmp_dir + 'snap%d' % i)
+            virsh.snapshot_create_as(self.vm.name, options,
+                                     ignore_status=False,
+                                     debug=True)
+            self.snap_path_list.append(self.tmp_dir + 'snap%d' % i)
+            self.snap_name_list.append('snap%d' % i)
+        qemu_img_cmd = 'qemu-img info --backing-chain %s' % self.snap_path_list[-1]
+        if libvirt_storage.check_qemu_image_lock_support():
+            qemu_img_cmd += " -U"
+        bc_info = process.run(qemu_img_cmd, verbose=True, shell=True).stdout_text
+
+        # Check chain
+        bc_result = self.check_backingchain(self.snap_path_list[::-1], bc_info)
+
+    def check_block_operation_result(self, vmxml, blockcommand, new_dev, bc_chain):
+        """
+        Run specific check backing chain function.
+
+        :param command: blockpull, blockcommit..
+        :param vmxml: vmxml obj of vm
+        :param dev: dev of target disk
+        :param bc_chain: original backing chain info as a list of source files
+
+        """
+
+        check_func = self.params.get('check_func', '')
+        check_bc_func_name = 'self.check_bc_%s' % check_func
+        check_bc = eval(check_bc_func_name)
+        #
+        if not callable(check_bc):
+            logging.warning('Function "%s" is not callable.', check_bc_func_name)
```

check_bc should be check_bc_func_name? And do you need to raise an exception?

---

This exception sentence comes from original blockcommand.py ,If not necessary ,Let me delete it now